app/autodiscovery/src/test/java/com/fsck/k9/autodiscovery/providersxml/ProvidersXmlDiscoveryTest.kt

```diff
@@ -28,23 +28,24 @@ class ProvidersXmlDiscoveryTest : RobolectricTest() {
         val connectionSettings = providersXmlDiscovery.discover("user@gmail.com")
 
         assertThat(connectionSettings).isNotNull()
-        with(connectionSettings!!.incoming) {
+        with(connectionSettings!!.incoming!!.first()) {
```

Comment:
Tests need updating to reflect type change. This is assuming there is only one result, we could just assert this.